cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js

```diff
@@ -136,7 +138,56 @@ class HydratorPlusPlusNodeConfigCtrl {
       }
     });
 
+    this.EventPipe.on('content-filter.changed', (widgetName, widgetValue) => {
+      this.contentFilterWidgetWiseLastEvent[widgetName] = widgetValue;
+    });
+
+    this.EventPipe.on('content-filter.destroyed', (widgetName) => {
+      if(this.contentFilterWidgetWiseLastEvent.hasOwnProperty(widgetName)) {
+        delete(this.contentFilterWidgetWiseLastEvent[widgetName]);
+      }
+    });
+
   }
+
+  shouldHideWidgetBasedOnFilter(widget) {
+    if(widget.hasOwnProperty('widget-attributes') && widget['widget-attributes'].hasOwnProperty('filters')) {
+      try {
+        // By default no need to hide.
+        let matchingFilterFound = false;
```

Comment:
is it needed ? because at last we return false by default.

Reply:
yes its values change later in the code.